tests/Model/BSONDocumentTest.php

```diff
@@ -4,10 +4,10 @@
 
 use MongoDB\Model\BSONArray;
 use MongoDB\Model\BSONDocument;
-use MongoDB\Tests\FunctionalTestCase;
+use MongoDB\Tests\TestCase;
 use ArrayObject;
 
-class BSONDocumentTest extends FunctionalTestCase
+class BSONDocumentTest extends TestCase
```

Comment:
The `testClone()` method from BSONArrayTest can be ported over to this class as well:

```php
$document = new BSONDocument([
    'a' => [
        'a' => 'foo',
        'b' => new stdClass,
        'c' => ['bar', new stdClass],
    ],
    'b' => new BSONDocument([
        'a' => 'foo',
        'b' => new stdClass,
        'c' => ['bar', new stdClass],
    ]),
]);
```